Core.lua

```diff
@@ -1990,12 +1990,14 @@ function Hekili.Update( initial )
                 UI.EventPayload = {}
             end
 
-            for k, v in pairs( Queue[ 1 ] ) do
-                UI.EventPayload[ k ] = v
-            end
-
-            if WeakAuras and WeakAuras.ScanEvents then
-                WeakAuras.ScanEvents( "HEKILI_RECOMMENDATION_UPDATE", dispName, Queue[ 1 ].actionID, Queue[ 1 ].indicator, Queue[ 1 ].empower_to, UI.EventPayload )
+            for i = 1, numRecs do
+                UI.EventPayload[ i ] = {}
+                for k, v in pairs( Queue[ i ] ) do
+                    UI.EventPayload[ i ][ k ] = v                   
+                end
+                if WeakAuras and WeakAuras.ScanEvents then
+                    WeakAuras.ScanEvents( "HEKILI_RECOMMENDATION_UPDATE", dispName, Queue[ i ].actionID, Queue[ i ].indicator, Queue[ i ].empower_to, UI.EventPayload[ i ] )
```

Comment:
There was a couple of ways to resolve this ultimately. 

One way would be to send over the entire UI.EventPayload table and fire the event one time. However, this could be seen as breaking the existing event contract as the payload table would no longer be returning a single slot.

I chose to fire the event multiple times, once for each recommendation as that appeared to be the most useful and the original intent. It also ensures the event parameters roughly line up with the Hekili_GetRecommendedAbility returns.